### PR TITLE
Cleanup all isV14 switches

### PR DESCRIPTION
**Pull request by myarmolinsky**
Cleanup all isV14 switches

Resolves: #2503 
Change-type: major
Signed-off-by: Matthew Yarmolinsky <matthew-timothy@balena.io>

---
Please check the CONTRIBUTING.md file for relevant information and some
guidance. Keep in mind that the CLI is a cross-platform application that runs
on Windows, macOS and Linux. Tests will be automatically run by balena CI on
all three operating systems, but this will only help if you have added test
code that exercises the modified or added feature code.

Note that each commit message (currently only the first line) will be
automatically copied to the CHANGELOG.md file, so try writing it in a way
that describes the feature or fix for CLI users.

If there isn't a linked issue or if the linked issue doesn't quite match the
PR, please add a PR description to explain its purpose or the features that it
implements. Adding PR comments to blocks of code that aren't self explanatory
usually helps with the review process.

If the PR introduces security considerations or affects the development, build
or release process, please be sure to highlight this in the PR description.

Thank you very much for your contribution!

### CODE REVIEW

File: lib/utils/common-flags.ts [hunk 1]
```diff
@@ -97,18 +96,7 @@ export const deviceType = flags.string({
...
-		: {
-				type: flags.string({
-					description: 'ignored - no longer required',
-					char: 't',
-					required: false,
-					hidden: true,
-				}),
-		  }),
-};
+export const deviceTypeIgnored = {};
```
Comment (thgreasi) [inline lib/utils/common-flags.ts:99]: Can we completely drop this, or is it used anywhere else?

File: docs/balena-cli.md [hunk 2]
```diff
@@ -333,6 +333,30 @@ Examples:
 
 ### Options
 
+#### --fields FIELDS
```
Comment (thgreasi) [inline docs/balena-cli.md:336]: Maybe we need to split these in separate commits to communicate to users what changed (rather than the not-that-useful "Cleanup all isV14 switches"). I'm guessing that using "blame" on GH to find the PR that added those version checks should help find a better commit message.

File: lib/utils/common-flags.ts (hunk 1 above)
Comment (myarmolinsky) [reply lib/utils/common-flags.ts:99]: It's used but it's pointless to keep, just doing `...{}` pretty much, dropping

File: docs/balena-cli.md [hunk 3]
```diff
@@ -367,6 +367,14 @@ fleet name, slug (preferred), or numeric ID (deprecated)
 
 open fleet dashboard page
 
+#### --fields FIELDS
+
+only show provided fields (comma-separated)
```
Comment (thgreasi) [inline docs/balena-cli.md:396]: Let's change the commit to `fleet: Add --fields and --json options`, to be consistent w/ how the changelog looks.

File: docs/balena-cli.md [hunk 4]
```diff
@@ -333,6 +333,30 @@ Examples:
...
+
+#### -j, --json
+
+output in json format
+
+#### --filter FILTER
+
+filter results by substring matching of a given field, eg: --filter field=foo
+
+#### --no-header
```
Comment (thgreasi) [inline docs/balena-cli.md:348]: Let's change the title to `fleets: Add --filter, --no-header, --no-truncate, and --sort options` so that we are consistent with the older changelog entries.

**Review by thgreasi — COMMENTED** [review]
It would be good to also mention the oclif output change on its own, as a heads-up that this might break output parsing. Maybe we could have a single commit that changes fleet & fleets to use this.outputData w/o passing the options, and we then add the options as extras?
